packages/react-devtools-extensions/src/background.js

```diff
@@ -63,19 +90,35 @@ function doublePipe(one, two) {
 }
 
 function setIconAndPopup(reactBuildType, tabId) {
-  chrome.browserAction.setIcon({
-    tabId: tabId,
-    path: {
-      '16': 'icons/16-' + reactBuildType + '.png',
-      '32': 'icons/32-' + reactBuildType + '.png',
-      '48': 'icons/48-' + reactBuildType + '.png',
-      '128': 'icons/128-' + reactBuildType + '.png',
-    },
-  });
-  chrome.browserAction.setPopup({
-    tabId: tabId,
-    popup: 'popups/' + reactBuildType + '.html',
-  });
+  if (IS_FIREFOX) {
+    chrome.browserAction.setIcon({
+      tabId: tabId,
+      path: {
+        '16': chrome.runtime.getURL(`icons/16-${reactBuildType}.png`),
+        '32': chrome.runtime.getURL(`icons/32-${reactBuildType}.png`),
+        '48': chrome.runtime.getURL(`icons/48-${reactBuildType}.png`),
+        '128': chrome.runtime.getURL(`icons/128-${reactBuildType}.png`),
+      },
+    });
+    chrome.browserAction.setPopup({
+      tabId: tabId,
+      popup: chrome.runtime.getURL(`popups/${reactBuildType}.html`),
+    });
```

Comment:
Instead of repeating code here, maybe we can instead do something like

```
const browserAction = IS_FIREFOX ? chrome.browserAction : chrome.action;
...
```

Reply:
Sure. When I coded this I thought it would be temporary but it turns out Firefox is pretty behind in V3 and this might stay for quite a while